File: dart/bob/lib/bob.dart
RegExp upper = RegExp(r'\p{Uppercase}', unicode: true);
RegExp lower = RegExp(r'\p{Lowercase}', unicode: true);

class Bob {
  String response(String input) {
    var trimmed = input.trimRight();

    var isSilence = trimmed.isEmpty;
    var isYelling = trimmed.contains(upper) && !trimmed.contains(lower);
    var isQuestion = trimmed.endsWith('?');

    /* switch _expressions_ require Dart v3:
     * ensure the sdk environment constraint is altered in pubspec.yaml
     */
    return switch ((isSilence, isYelling, isQuestion)) {
      (true, _, _) => 'Fine. Be that way!',
      (_, true, true) => "Calm down, I know what I'm doing!",
      (_, true, false) => 'Whoa, chill out!',
      (_, false, true) => 'Sure.',
      _ => 'Whatever.',
    };
  }
}

File: php/bob/Bob.php
<?php

declare(strict_types=1);

class Bob
{
    public function respondTo(string $str): string
    {
        $cleaned = rtrim($str);
        $isSilence = $cleaned == '';
        $isQuestion = str_ends_with($cleaned, '?');
        $isYelling = preg_match('/[[:upper:]]/', $str)
                 && !preg_match('/[[:lower:]]/', $str);

        if ($isSilence) {
            return 'Fine. Be that way!';
        } elseif ($isYelling && $isQuestion) {
            return "Calm down, I know what I'm doing!";
        } elseif ($isYelling) {
            return 'Whoa, chill out!';
        } elseif ($isQuestion) {
            return 'Sure.';
        } else {
            return 'Whatever.';
        }
    }
}
